libraries/libfc/src/crypto/elliptic_secp256k1.cpp

```diff
@@ -79,6 +81,14 @@ namespace fc { namespace ecc {
       return fc::sha512::hash( serialized_result.begin() + 1, serialized_result.size() - 1 );
     }
 
+    private_key private_key::generate()
+    {
+       private_key ret;
+       do {
+         RAND_bytes((uint8_t*)ret.my->_key.data(), sizeof(ret.my->_key.data()));
+       } while(!secp256k1_ec_seckey_verify(detail::_get_context(), (const uint8_t*)ret.my->_key.data()));
+       return ret;
+    }
```

Comment:
This change requires special attention during review. BoringSSL does not support the K1 curve. We've previously transitioned to using libsecp256k1 for all things K1... except for private key generation as performed here.

`_key` is a `private_key_secret`,
https://github.com/AntelopeIO/leap/blob/409eb5960c09b8b539984d2f098cf746be48b640/libraries/libfc/src/crypto/_elliptic_impl_priv.hpp#L22

and `private_key_secret` is a `fc::sha256`,
https://github.com/AntelopeIO/leap/blob/409eb5960c09b8b539984d2f098cf746be48b640/libraries/libfc/include/fc/crypto/elliptic.hpp#L23

The original code in this PR is probably incorrect and does not generate 32 random bytes, which would be a disaster.